### PR TITLE
Codespell data and doc dirs

### PR DESCRIPTION
**Pull request by matthiaskrgr**
run codespell on data and doc directory (src is in progress)
https://github.com/lucasdemarchi/codespell

### CODE REVIEW

File: data/campaigns/Dead_Water/scenarios/01_Invasion.cfg [hunk 1]
```diff
@@ -17,7 +17,7 @@
 # under control. When it is, bring in more units to help them. Attack
 # during the day, but retreat and heal at night: You can't afford to
 # have very many units zombified. Also, rotate your units and use the
-# ZOC of Gwabbo and your level ones to sheild your wounded. Fighters
+# ZOC of Gwabbo and your level ones to shield your wounded ones. Fighters
```
Comment (jyrkive) [inline data/campaigns/Dead_Water/scenarios/01_Invasion.cfg]: "Ones" isn't necessary.

File: data/core/help.cfg [hunk 2]
```diff
@@ -547,7 +547,7 @@ Switch debug mode on (does not work in multiplayer). See <ref>dst='debug_command
 Debug mode is turned off by quitting the game or :nodebug.
 
 " + "<header>text=':droid [<side>] [on|off]'</header>" + _ "
-Set or toggle player on side between human and AI player. The player/client who controls that side needs to issue this command. If no second parameter is supplied, toggle bewteen human and AI. If it is ‘on’, set an AI controller. If it is ‘off’ set a human controller. Defaults to the currently active side if no parameter is supplied.
+Set or toggle player on side between human and AI player. The player/client who controls that side needs to issue this command. If no second parameter is supplied, toggle between human and AI. If it is ‘on’, set an AI controller. If it is ‘off’ set a human controller. Defaults to the currently active side if no parameter is supplied.
```
Comment (jyrkive) [inline data/core/help.cfg]: Translatable. Please add to pofix.

File: data/core/units/merfolk/Entangler.cfg [hunk 3]
```diff
@@ -16,7 +16,7 @@
     {AMLA_DEFAULT}
     cost=46
     usage=mixed fighter
-    description= _ "Merfolk tend to use a combination of both nets and clubs in fishing; nets to collect the fish, and clubs to finish the kill. Both of these tools have seen adaptation for warfare, and are valuable enough that auxilliaries who specialize in using them are deliberately brought to battle.
+    description= _ "Merfolk tend to use a combination of both nets and clubs in fishing; nets to collect the fish, and clubs to finish the kill. Both of these tools have seen adaptation for warfare, and are valuable enough that auxiliaries who specialize in using them are deliberately brought to battle.
```
Comment (jyrkive) [inline data/core/units/merfolk/Entangler.cfg]: Translatable. Please add to pofix.

File: data/gui/window/addon_manager.cfg [hunk 4]
```diff
@@ -691,7 +691,7 @@
 					id = "filter"
 					definition = "filter"
 
-					tooltip = _ "Filters on addon descripton, version, type or author."
+					tooltip = _ "Filters on addon description, version, type or author."
```
Comment (jyrkive) [inline data/gui/window/addon_manager.cfg]: Translatable. Please add to pofix.

File: data/lua/ilua.lua [hunk 5]
```diff
@@ -4,7 +4,7 @@
 -- will try to print out tables recursively, subject to the pretty_print_limit value.
 -- Steve Donovan, 2007
 -- Adapted by iceiceice for wesnoth, 2014
--- Retrived from: http://lua-users.org/files/wiki_insecure/users/steved/ilua.lua
+-- Retrieeved from: http://lua-users.org/files/wiki_insecure/users/steved/ilua.lua
```
Comment (jyrkive) [inline data/lua/ilua.lua]: Typo. Should be "retrieved". :P

File: data/campaigns/Dead_Water/scenarios/01_Invasion.cfg (hunk 1 above)
Comment (Vultraz) [reply data/campaigns/Dead_Water/scenarios/01_Invasion.cfg]: Seconded.

**Comment by matthiaskrgr** [comment]
What does "pofix" mean?

**Comment by jyrkive** [comment]
@matthiaskrgr https://github.com/wesnoth/wesnoth/blob/master/utils/pofix.py

File: data/campaigns/Dead_Water/scenarios/01_Invasion.cfg (hunk 1 above)
Comment (CelticMinstrel) [reply data/campaigns/Dead_Water/scenarios/01_Invasion.cfg]: Thirded.

File: data/ai/micro_ais/cas/ca_protect_unit_attack.lua [hunk 6]
```diff
@@ -48,7 +48,7 @@ function ca_protect_unit_attack:evaluation(cfg)
                     if counter_damage_table[str] then  -- If so, use saved value
                         max_counter_damage = max_counter_damage + counter_damage_table[str]
                     else  -- if not, calculate it and save value
-                        -- Go thru all weapons, as "best weapon" might be different later on
+                        -- Go through all weapons, as "best weapon" might be different later on
```
Comment (CelticMinstrel) [inline data/ai/micro_ais/cas/ca_protect_unit_attack.lua]: This change seems pointless, as "thru" is pretty accepted now in an informal context.

File: data/ai/micro_ais/scenarios/messenger_escort.cfg [hunk 7]
```diff
@@ -118,7 +118,7 @@
 
         # waypoints for AI
         {PLACE_IMAGE "scenery/signpost.png" 31 20}
-        {SET_LABEL 31 20 _"Messanger Waypoint 1"}
+        {SET_LABEL 31 20 _"Messenger Waypoint 1"}
```
Comment (CelticMinstrel) [inline data/ai/micro_ais/scenarios/messenger_escort.cfg]: Another translatable that would need to be pofixed. Though, given the timing, I dunno, maybe not...

File: utils/pofix.py [hunk 8]
```diff
@@ -83,6 +85,7 @@
 ("displayed in the upper right, (under the default theme)", "displayed in the upper right (under the default theme)"),
 ("However the few implements", "However, the few implements"),
 ("humans can learn to wield it and able to learn", "humans can learn to wield it and are able to learn"),
+("If no second parameter is supplied, toggle bewteen human and AI.", "If no second parameter is supplied, toggle between human and AI."),
```
Comment (jyrkive) [inline utils/pofix.py]: The lines in pofix are intended to be only short snippets, not entire strings (except if the string itself is only a few words).

Comment (matthiaskrgr) [reply utils/pofix.py]: ok, I shortened them.

**Pull request merged**